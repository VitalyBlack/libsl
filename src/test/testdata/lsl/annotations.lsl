libsl "1.0.0";
library simple;

types {
   Int(int);
}

automaton A : Int {
   fun f(@annotation param: Int);
}
