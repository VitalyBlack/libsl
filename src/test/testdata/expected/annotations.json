{
  "name": "simple",
  "lslVersion": "1.0.0",
  "types": [
    {
      "name": "Int",
      "isPointer": false,
      "kind": "alias",
      "originalType": {
        "name": "int32",
        "isPointer": false
      }
    }
  ],
  "automata": [
    {
      "name": "A",
      "type": "Int",
      "constructorVariables": [],
      "variables": [],
      "states": [],
      "shifts": [],
      "functions": [
        {
          "name": "f",
          "automaton": "A",
          "target": "A",
          "args": [
            {
              "name": "param",
              "type": "Int"
            }
          ],
          "contracts": [],
          "statements": []
        }
      ]
    }
  ]
}